script.js

```diff
@@ -0,0 +1,70 @@
+// Configura el tema inicial usando el tema guardado o el del sistema
+const savedTheme = localStorage.getItem('theme') || (window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light');
+setTheme(savedTheme);
+
+// Detecta cambios en la preferencia de tema del sistema y aplica el cambio
+window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', (event) => {
+    setTheme(event.matches ? 'dark' : 'light');
+});
+
+// Cambia y guarda el tema
+function toggleTheme() {
+    const currentTheme = document.documentElement.getAttribute('data-theme');
+    const newTheme = currentTheme === 'dark' ? 'light' : 'dark';
+    setTheme(newTheme);
+}
+
+// Aplica el tema y lo guarda en localStorage
+function setTheme(mode) {
+    document.documentElement.setAttribute('data-theme', mode);
+    localStorage.setItem('theme', mode);
+}
+
+
+// Maneja el formulario de nota
+document.getElementById('note-form').addEventListener('submit', function(e) {
```

Comment:
Avoid mental mapping.
```suggestion
document.getElementById('note-form').addEventListener('submit', function(event) {
```

- [ ] Rename `e` as `event`